manticore/core/manticore.py

```diff
@@ -1139,3 +1178,41 @@ def save_run_data(self):
             config.save(f)
 
         logger.info("Results in %s", self._output.store.uri)
+
+    @at_not_running
+    def set_instrospection_plugin(self, new_plugin: type):
+        """
+        Allows the user to extend the functionality of the default IntrospectionAPI plugin.
+        Must be called before ManticoreBase.run()
+
+        :param new_plugin: a subclass of IntrospectionAPIPlugin
+        """
+        assert issubclass(
+            new_plugin, self._introspection_plugin
+        ), "New plugin must be a subclass of IntrospectionAPIPlugin"
+        self._introspection_plugin = new_plugin
+
+    def introspect(self) -> typing.Dict[int, StateDescriptor]:
+        """
+        Allows callers to view descriptors for each state
+
+        :return: the latest copy of the State Descriptor dict
+        """
+        if self._introspector is not None:
+            return self._introspector.get_state_descriptors()
```

Comment:
So if we have a way to name the plugins can not this then translate into something like:
```
...
m.plugins['introspector'].get_state_descriptor()
...
```
Also do you imagine that to be called from any place other than the main script?

Reply:
Yes, that's exactly what I intend to translate this to in #1781.

> Also do you imagine that to be called from any place other than the main script?

In the TUI branch, it's called from the server thread, and I use it in several of the daemon thread examples. I don't think it makes sense to call it from a plugin or worker, but I also don't think that would hurt anything. Unless accessing the locked context from a worker could cause a deadlock?

Reply:
Need to find out why that server thread or whoever is calling `introspect()` does not have a reference to the introspector plugin already.